plugins/inputs/mikrotik/parsers.go

```diff
@@ -0,0 +1,103 @@
+package mikrotik
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+)
+
+func parse(data common) (points []parsedPoint, err error) {
+	var errorList []string
+
+	data = filterCommon(data)
+
+	for i := range data {
+		tags := make(map[string]string)
+		fields := make(map[string]interface{})
+		for _, tagName := range tagFields {
+			if data[i][tagName] != "" {
+				tags[tagName] = data[i][tagName]
+			}
+		}
+		for _, fieldName := range valueFields {
+			if data[i][fieldName] != "" {
+				if slices.Contains(durationParseFieldNames, fieldName) {
+					fields[fieldName], err = parseUptimeIntoDuration(data[i][fieldName])
+					if err != nil {
+						errorList = append(errorList, err.Error())
+					}
+				} else if strings.Contains(data[i][fieldName], ",") {
+					rxTxValues := strings.Split(data[i][fieldName], ",")
+					fields[fieldName+"_tx"], err = strToInt64(rxTxValues[0])
+					if err != nil {
+						errorList = append(errorList, err.Error())
+					}
+					fields[fieldName+"_rx"], err = strToInt64(rxTxValues[1])
+					if err != nil {
+						errorList = append(errorList, err.Error())
+					}
+				} else {
+					fields[fieldName], err = strToInt64(data[i][fieldName])
+					if err != nil {
+						errorList = append(errorList, err.Error())
+					}
+				}
+			}
+		}
+
+		points = append(points, parsedPoint{Tags: tags, Fields: fields})
```

Comment:
You could create the metric here, can't you? This way you can get rid of the extra struct and/or directly add them to the accumulator.